### PR TITLE
create trackEvent and identifyUser browser-destinations for 1flow

### PR DESCRIPTION
**Pull request by namit1Flow**
**1Flow web**
- In this repo we are targeting trackEvent and identifyUser.
- Initialize 1Flow and fire trackEvent(track) and idenfiyUser(identifyUser)

### CODE REVIEW

**Comment by namit1Flow** [comment]
please let me know if any changes you want.
Thanks

**Comment by joe-ayoub-segment** [comment]
hi @namit1Flow - thanks for raising this PR. 

As this is for a new Integration would you be available to meet with me to help me understand the use-cases for the Integration, as well as the types of data you want to send to the 1Flow platform?

You can schedule a call with me here: http://calendly.com/joe_ayoub/integration-workshop

I look forward to hearing from you!

Best regards,
Joe

**Comment by namit1Flow** [comment]
Hi Joe,

We have a meeting today at 10:00 PM IST to discuss the integration.

**Comment by joe-ayoub-segment** [comment]
Hi Namit, 

Thanks for taking the time to catch up. 
Please see the diff file for recommended changes, and please add unit tests. 

Kind regards,
Joe

**Comment by joe-ayoub-segment** [comment]
Hi @namit1Flow , just following up on this PR since our meeting on 8-Nov. Are you still interested in getting this PR live?
Are you clear on the changes we discussing during the call (which were shared via the diff file)?

Best regards,
Joe

File: packages/browser-destinations/destinations/1flow/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+}
+
+export const destination: BrowserDestinationDefinition<Settings, _1Flow> = {
+  name: '1Flow',
+  slug: 'actions-1flow',
+  mode: 'device',
+
+  settings: {
+    projectApiKey: {
+      description: 'The app_id of your 1Flow app which will indicate where to store any data.',
```
Comment (joe-ayoub-segment) [inline packages/browser-destinations/destinations/1flow/src/index.ts:22]:
```suggestion
      description: 'Your 1Flow app_id - indicates where in 1Flow send Segment events to.',
```

File: packages/browser-destinations/destinations/1flow/src/identifyUser/index.ts [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+      required: false,
+      default: {
+        '@path': '$.traits.email'
+      }
+    }
+  },
+  perform: (_1Flow, event) => {
+    const { userId, anonymousId, traits, first_name, last_name, phone, email } = event.payload
+    _1Flow('identify', userId, anonymousId, {
+      traits: traits,
```
Comment (joe-ayoub-segment) [inline packages/browser-destinations/destinations/1flow/src/identifyUser/index.ts:81]: note: you might want to dedupe first_name, last_name, email and phone from the traits object.

File: packages/browser-destinations/destinations/1flow/src/trackEvent/generated-types.ts [hunk 3]
```diff
@@ -0,0 +1,20 @@
+export interface Payload {
```
Comment (joe-ayoub-segment) [inline packages/browser-destinations/destinations/1flow/src/trackEvent/generated-types.ts:1]: please run this command to auto generate the types files: 

./bin/run generate:types

**Review by joe-ayoub-segment — CHANGES_REQUESTED** [review]

**Comment by namit1Flow** [comment]
Hi, @joe-ayoub-segment  suggested issues I fixed. Please check.

File: packages/browser-destinations/destinations/1flow/src/index.ts [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+  interface Window {
+    _1Flow: _1Flow
+  }
+}
+
+export const destination: BrowserDestinationDefinition<Settings, _1Flow> = {
+  name: '1Flow',
+  slug: 'actions-1flow',
+  mode: 'device',
+
```
Comment (joe-ayoub-segment) [inline packages/browser-destinations/destinations/1flow/src/index.ts:19]: please add description here: 

description: 'Send user analytics events and user profile data from Segment to 1Flow',

**Review by joe-ayoub-segment — CHANGES_REQUESTED** [review]

**Review by joe-ayoub-segment — APPROVED** [review]

**Comment by joe-ayoub-segment** [comment]
Hi @namit1Flow - looks like the tests are failing. Can you take a look please?

image

**Comment by namit1Flow** [comment]
Hi @joe-ayoub-segment ,

Please check my latest commit and let me know if any action is needed.

**Review by joe-ayoub-segment — APPROVED** [review]

**Comment by joe-ayoub-segment** [comment]
hi @namit1Flow - this PR has been deployed. Can you please let me know if you are happy with the change?

Kind regards,
Joe

**Pull request merged**